src/components/VerticalNavigation/VerticalNavigationItem.js

```diff
@@ -0,0 +1,215 @@
+      iconStyleClass,
+      title,
+      badges,
+      showMobileSecondary,
+      showMobileTertiary,
+      navCollapsed,
+      handlePrimaryHover,
+      handleSecondaryHover,
+      handleTertiaryHover,
+      handlePrimaryBlur,
+      handleSecondaryBlur,
+      handleTertiaryBlur,
+      handlePrimaryClick,
+      handleSecondaryClick,
+      handleTertiaryClick,
+    } = this.props;
+
+    const handleHover = {
+      primary: this.handlePrimaryHover,
+      secondary: this.handleSecondaryHover,
+      tertiary: this.handleTertiaryHover,
+    };
+
+    const handleBlur = {
+      primary: this.handlePrimaryBlur,
+      secondary: this.handleSecondaryBlur,
+      tertiary: this.handleTertiaryBlur,
+    };
+
+    const handleClick = {
+      primary: this.handlePrimaryClick,
+      secondary: this.handleSecondaryClick,
+      tertiary: this.handleTertiaryClick,
+    };
+
+    // TODO maybe don't pass in all three handlers at the top, instead pass them below? would that still be DRY?
+
+    // Assume that if someone nests deeper than tertiary, it's just another tertiary-styled item.
+    const depth = this.props.depth || 'primary';
+    const nextDepth = VerticalNavigation.getNextDepth(depth);
+
+    // Default to collapsed unless we explicitly pass collapsed = false.
+    const secondaryCollapsed = this.props.secondaryCollapsed !== false;
+    const tertiaryCollapsed = this.props.tertiaryCollapsed !== false;
+
+    // We only have primary, secondary, and tertiary depths, so nextDepth will only ever be secondary or tertiary.
+    const nextDepthCollapsed =
+      nextDepth === 'secondary' ? secondaryCollapsed : tertiaryCollapsed;
+    const collapseNextNav =
+      nextDepth === 'secondary'
+        ? this.collapseSecondaryNav
+        : this.collapseTertiaryNav;
+
+    return (
+      <ListGroupItem
+        listItem // Renders as <li>. Other props can change this, see logic in react-bootstrap's ListGroupItem.
+        className={cx({
+          [`${nextDepth}-nav-item-pf`]:
+            depth !== 'tertiary' && children && children.length > 0,
+          active: trackActiveState, // This is the only class we have at the tertiary depth.
+          'visible-xs-block': trackActiveState, // TODO is trackActiveState the right thing here, and is it named right?
+          'is-hover': depth !== 'tertiary' && trackHoverState,
+          // This class is present at primary and secondary depths if mobileItem is true,
+          // except for the primary depth, where it is only present if showMobileSecondary is also true.
+          'mobile-nav-item-pf':
+            mobileItem &&
+            ((depth === 'primary' && showMobileSecondary) ||
+              depth === 'secondary'),
+          // This class is confusingly named, but the logic is more readable.
+          'mobile-secondary-item-pf':
+            mobileItem && depth === 'primary' && showMobileTertiary,
+          // I don't know, that's just how this stuff was in patternfly-ng...
```

Comment:
Yeah, could have been named better but basically it is used to handle mobile navigation where only one of the menus (primary, secondary, or tertiary) are shown at a time (the navigation is drill down in mobile mode).

Reply:
Makes sense. I'll change this comment to be a little less confused, haha.